content/en/platform/corda/4.3/enterprise/event-scheduling.md

```diff
@@ -75,6 +75,15 @@ method and if they do not return `null` then that activity is scheduled based on
 encounter but which aren’t related to yourself will not have any activities scheduled.
 
 
+## Scheduled events flow logic
+
+`SchedulableState` must return a fixed time based only upon data in the state. That is because any node that inserts the state into their vault, i.e. the state isRelevant, will run the code to calculate the scheduled time and put in a timer for the deadline returned (if non-null). This is re-done every time node restarts. When timer point is crossed (i.e. if node misses the precise second it is still going to happen), the flow names is started. It is the responsibility of that flow to consume and modify the state to change or cancel the timer. Thus, logic will always retry after restart.
```

Comment:
```suggestion
The `SchedulableState` returns a fixed time based only on the state data. That is because any node that inserts a `SchedulableState` state into their vault, such as the state `isRelevant`, runs the code to calculate the scheduled time. The node then starts a timer counting down to the deadline (if non-null). This process restarts every time the node restarts. When the timer's endpoint is crossed, the flow names are started. Even if the node misses the precise endpoint of the set fixed time, this will still happen. It is the responsibility of that flow to consume and modify the state to change or cancel the timer. Thus, logic will always retry after restart.
```

Reply:
Some suggestions, though especially on the last 2 sentence I'm not really sure I followed the logic correctly. So if any of my ideas make it technically incorrect apologies.